proto/v1beta1/intoto.proto

```diff
@@ -0,0 +1,145 @@
+// limitations under the License.
+
+syntax = "proto3";
+
+package grafeas.v1beta1.intoto;
+
+option go_package = "github.com/grafeas/grafeas/proto/v1beta1/intoto_go_proto";
+option java_multiple_files = true;
+option java_package = "io.grafeas.v1beta1.intoto";
+option objc_class_prefix = "GRA";
+
+import "google/protobuf/struct.proto";
+
+// This file defines the proto that matches the schema of an in-toto link
+// file. in-toto is a framework to protect a software supply chain. You can read
+// more about in-toto at https://in-toto.io/ and you can find the current
+// version of the in-toto specification at
+// https://github.com/in-toto/docs/blob/master/in-toto-spec.md
+
+
+// A signature object consists of the KeyID used and the signature itself.
+message Signature {
+  string keyid = 1 [json_name="keyid"];
+  string sig = 2 [json_name="sig"];
+}
+
+// An object defined to capture the by products of a step.
+message ByProductObject {
+  int32 return_value = 1 [json_name="return-value"];
+  string stderr = 2 [json_name="stderr"];
+  string stdout = 3 [json_name="stdout"];
+}
+
+// This corrresponds to the core in-toto link attestation.
+message Link {
+  // This field contains the full command executed for the step. This can also
+  // be empty if attestations are generated for operations that aren't directly
+  // mapped to a specific command. Each term in the command is an independent
+  // string in the list. An example of a command in the in-toto metadata field
+  // is:
+  // "command": ["git", "clone", "https://github.com/in-toto/demo-project.git"]
+  repeated string command = 1 [json_name="command"];
+
+  // Materials are the supply chain artifacts that go into the step and are used
+  // for the operation performed. The key of the map is the path of the artifact
+  // and the structure contains the recorded hash information. An example is:
+  // "materials": {
+  //   "foo/bar": {
+  //     "sha256": "ebebf..."
+  //   }
+  // }
+  map<string, google.protobuf.Struct> materials = 2 [json_name="materials"];
+
+  // Products are the supply chain artifacts generated as a result of the step.
+  // The structure is identical to that of materials.
+  map<string, google.protobuf.Struct> products = 3 [json_name="products"];
+
+  // By Products are made up of the data that is generated as part of a software
+  // supply chain step, but are not the actual result of the step.
+  ByProductObject byproducts = 4 [json_name="byproducts"];
+
+  // This is a field that can be used to capture information about the
+  // environment. It is recommended for this field to contain information that
+  // details environment variables, filesystem information, and the present
+  // working directory. The structure of this field is:
+  // "environment": {
+  //   "variables": "<ENV>",
+  //   "filesystem": "<FS>",
+  //   "workdir": "<CWD>"
+  // }
+  map<string, google.protobuf.Struct> environment = 5 [json_name="environment"];
```

Comment:
nit: here and everywhere, if `json_name` is the same as the field name, we don't need to include it, right?